bentoml/deployment/aws_lambda/__init__.py

```diff
@@ -0,0 +1,377 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+import logging
+from six.moves.urllib.parse import urlparse
+
+import boto3
+from packaging import version
+from ruamel.yaml import YAML
+
+from bentoml.bundler import loader
+from bentoml.deployment.aws_lambda.utils import (
+    ensure_sam_available_or_raise,
+    upload_bento_service_artifacts_to_s3,
+    init_sam_project,
+    lambda_deploy,
+    lambda_package,
+    call_sam_command,
+    check_s3_bucket_exists_or_raise,
+    validate_lambda_template,
+)
+from bentoml.deployment.operator import DeploymentOperatorBase
+from bentoml.deployment.utils import (
+    exception_to_return_status,
+    ensure_deploy_api_name_exists_in_bento,
+    ensure_docker_available_or_raise,
+)
+from bentoml.exceptions import BentoMLException
+from bentoml.proto.deployment_pb2 import (
+    ApplyDeploymentResponse,
+    Deployment,
+    DeploymentState,
+    DescribeDeploymentResponse,
+    DeleteDeploymentResponse,
+)
+from bentoml.proto.repository_pb2 import GetBentoRequest, BentoUri
+from bentoml.utils import Path
+from bentoml.utils.s3 import is_s3_url
+from bentoml.utils.tempdir import TempDirectory
+from bentoml.yatai.status import Status
+
+
+logger = logging.getLogger(__name__)
+
+
+def generate_function_resource(
```

Comment:
probably easier to read just move this into  `_create_aws_lambda_cloudformation_template_file ` instead of a separate function?